pandas/core/indexes/base.py

```diff
@@ -3763,18 +3763,59 @@ def _validate_can_reindex(self, indexer: np.ndarray) -> None:
 
     def reindex(self, target, method=None, level=None, limit=None, tolerance=None):
         """
-        Create index with target's values.
+        Create an index with target's values.
 
         Parameters
         ----------
         target : an iterable
+        method : {None, ‘backfill’/’bfill’, ‘pad’/’ffill’, ‘nearest’}
```

Comment:
Is there a way to share parts with NDFrame.reindex?

Reply:
Maybe share parts with other functions in this same file, such as get_indexer? @phofl